frontend/src/metabase/lib/utils.js

```diff
@@ -143,11 +143,15 @@ const MetabaseUtils = {
     );
   },
 
-  validEmail(email) {
+  isEmail(email) {
     const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
     return re.test(email);
   },
 
+  isInteger(string) {
```

Comment:
Maybe `isIntegerString`, or add a type check in case an actual number is passed. WDYT?

Reply:
How much re-writing would it require for us to use `input type ="number"` for numeric fields globally?
This would solve many bugs and we'll have "client-side validation" for free.

Reply:
Can't say for globally, but it's trivial to make numeric fields in `Settings` tab to render as `type="number"`. Do you mind these arrows?

<img width="436" alt="Screen Shot 2021-07-20 at 8 22 30 pm" src="https://user-images.githubusercontent.com/8542534/126368450-2f6edc1e-7d29-4cf4-8512-2e73aabcfe57.png">

Reply:
I don't mind them at all.
But that's a very good question for @mazameli or @brunobergher .